src/put-file.js

```diff
@@ -39,12 +39,19 @@ module.exports = async (client, storage, fileKey, file, opts = {}) => {
       copyFile = { headers: fileInfo };
     } else {
       copyFile = await new Promise((resolve, reject) => {
-        storage.fetch(sourceKey, { acl: 'private' }, (err, headers, buffer) => {
+        let decodedSourceKey;
+        try {
+          decodedSourceKey = sourceKey.split('/').map(decodeURIComponent).join('/');
+        } catch {
```

Comment:
This has a high Node requirement. Please include `catch (ex /* ignore */)` for now.